webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/ncf/TestTechniqueCompilationCache.scala

```diff
@@ -150,13 +153,25 @@ class TestTechniqueCompilationCache extends Specification with BeforeAfterAll {
       .runNow
   }
 
+  val expectedListOfOutputs: List[EditorTechniqueResult] = List(
+    EditorTechniqueResult(technique1.id, technique1.version, technique1.name, CompilationResult.Error("tech1 error")),
+    EditorTechniqueResult(technique2.id, technique2.version, technique2.name, CompilationResult.Success),
+    EditorTechniqueResult(technique3.id, technique3.version, technique3.name, CompilationResult.Error("tech3 error"))
+  )
+
   override def afterAll(): Unit = {
     compilationDir.delete()
   }
 
   "ReadTechniqueCompilationStatusService" should {
     "read with cumulated errors" in {
-      compilationStatusService.get().runNow must beEqualTo(
+      compilationStatusService.get().runNow must beEqualTo(expectedListOfOutputs)
+    }
+  }
```

Comment:
with that structure, it's easier to decouple and test gathering data separactly form saving. 
We should also test that the actor get what we want.

Reply:
OK I'll test it when the PR is merged !